tests/test_multi_thread.cpp

```diff
@@ -211,7 +211,6 @@ TEST_CASE("HNSW Multi-threading", "[ft][hnsw]") {
                 ->Float32Vectors(data.get() + i * dim)
                 ->Owner(false);
             auto add_res = index->Add(dataset);
-            REQUIRE(add_res.has_value());
```

Comment:
REQUIRE is not thread-safe. If add_res.has_value() is false, then add_res.value().size() will cause a core dump.